purepyvmware/datastores.py

```diff
@@ -0,0 +1,329 @@
+
+        Returns:
+            datastores (vim.view.ContainerView): Object view of datastore resources.
+        """
+        datastore_container = self.vsphere_content.viewManager.CreateContainerView(self.vsphere_content.rootFolder,
+                                                                                   [vim.Datastore], True)
+        datastores = datastore_container.view
+        datastore_container.Destroy()
+
+        return datastores
+
+    def get_all_pure_datastores(self):
+        """Retrieve all datastores backed by Pure Storage FlashArray volumes.
+
+        Returns:
+            pure_datastores (list): List of vim.Datastore objects associated with Pure Storage FlashArray(s).
+        """
+        pure_datastores = []
+        all_datastores = self.create_datastore_container_view()
+
+        for datastore in all_datastores:
+            devices = datastore_utils.get_datastore_identifier(datastore)
+            for device in devices:
+                dev_match = datastore_utils.compare_identifier(device)
+                if dev_match:
+                    pure_datastores.append(datastore)
+
+        return pure_datastores
+
+    def rescan_esxi_storage(self, esxi_hosts):
+        """Rescan all storage HBAs on selected ESXi hosts.
+
+        Args:
+            esxi_hosts (list): ESXi hosts to perform a storage rescan on.
+        """
+        for esxi_host in esxi_hosts:
+            esxi_host.configManager.storageSystem.RescanAllHba()
+
+    def verify_vsphere_cluster(self, cluster_name, flasharray):
+        """Verify host objects and host groups associated with the vSphere Cluster are configured on the FlashArray.
+
+        Args:
+            cluster_name (str): Name of the vSphere Cluster to verify.
+            flasharray (purestorage.FlashArray): Existing session to FlashArray.
+
+        Returns:
+            hgroup (dict): Host group associated with vSphere cluster.
+            connected_esxi_hosts (list): List of ESXi hosts in a connected state for vSphere cluster..
+        """
+        msg = 'Cancelling datastore creation. Please verify the object exists ,and is online, then try again.'
+
+        cluster = None
+        connected_esxi_hosts = []
+        esxi_hosts = []
+
+        compute_resources = self.create_compute_container_view()
+        for compute in compute_resources:
+            if compute.name == cluster_name:
+                cluster = compute
+
+        if cluster:
+            hgroup = datastore_utils.fa_hgroup_to_vc_cluster_mapping(cluster, flasharray)
+            esxi_hosts = cluster.host
+        else:
+            raise ValueError(f'Cluster \'{cluster_name}\' not found on specified vCenter Server. {msg}')
+
+        if esxi_hosts:
+            for esxi in esxi_hosts:
+                if esxi.runtime.connectionState == 'connected':
+                    connected_esxi_hosts.append(esxi)
+        else:
+            raise ValueError(f'No ESXi hosts found for \'{cluster_name}\'. {msg}')
```

Comment:
This is fine, however you can avoid needing the escapes if you used double-quotes around this text or around the internal quotations.

i.e.

`f"No ESXi hosts found for '{cluster_name}'."`

`f'No ESXi hosts found for "{cluster_name}".'`